src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -295,15 +296,34 @@ public void run() {
                 fieldName);
 
         type = field.getType();
-        accessors.add(context.getNamer().getFieldGetFunctionName(field));
+        accessors.add(
+            AccessorView.MemberView.newBuilder()
+                .member(context.getNamer().getFieldGetFunctionName(field))
+                .build());
       } else if (token == '[') {
         Preconditions.checkArgument(
             type.isRepeated() && !type.isMap(),
             "%s:%s: %s is not a repeated field",
             context.getMethodModel().getSimpleName(),
             valueSet.getId(),
             config.input());
-        throw new UnsupportedOperationException("array indexing not supported yet");
+        Preconditions.checkArgument(
+            config.scan() == Scanner.INT,
+            "%s:%s: expected int in index expression: %s",
+            context.getMethodModel().getSimpleName(),
+            valueSet.getId(),
+            config.input());
+
+        accessors.add(AccessorView.IndexView.newBuilder().index(config.tokenStr()).build());
+
+        Preconditions.checkArgument(
+            config.scan() == ']',
+            "%s:%s: expected ']': %s",
+            context.getMethodModel().getSimpleName(),
+            valueSet.getId(),
+            config.input());
+      } else if (token == '{') {
```

Comment:
since `.index` is a string, extending to maps should be pretty easy over here, correct?
(when we get to it, I suspect we'll want to s/`index`/`key`/ for generality)

Reply:
Yup it should be. I know of languages that treat array lookup and map lookup differently though (Perl comes to mind), I think we should treat them differently at this stage.